mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -0,0 +1,200 @@
+        pKVTemp->LoadFromFile(g_pFullFileSystem, SPEEDOMETER_DEFAULT_FILENAME, "MOD");
+        pKVTemp->SaveToFile(g_pFullFileSystem, SPEEDOMETER_FILENAME, "MOD");
+        pKVTemp->deleteThis();
+    }
+
+    m_pGamemodeSetupData->LoadFromFile(g_pFullFileSystem, SPEEDOMETER_FILENAME, "MOD");
+
+    KeyValues *pGamemodeKV = m_pGamemodeSetupData->FindKey(g_szGameModes[gametype]);
+    if (pGamemodeKV) // game mode is known
+    {
+        m_CurrentlyLoadedGamemodeSettings = gametype;
+
+        // get ordering
+        KeyValues *pOrderKV = pGamemodeKV->FindKey("order");
+        if (pOrderKV)
+        {
+            auto pOrderList = g_pSpeedometer->GetLabelOrderListPtr();
+            pOrderList->RemoveAll();
+            char szIndex[BUFSIZELOCL];
+            for (unsigned int i = 1; i <= SPEEDOMETER_MAX_LABELS; i++)
+            {
+                Q_snprintf(szIndex, BUFSIZELOCL, "%i", i);
+                const char *szSpeedo = pOrderKV->GetString(szIndex);
+                if (szSpeedo)
+                {
+                    SpeedometerLabel *label = GetLabelFromName(szSpeedo);
+                    if (label)
+                    {
+                        pOrderList->AddToTail(label);
+                    }
+                }
+            }
+        }
+        else
+        {
+            g_pSpeedometer->ResetLabelOrder();
+        }
+
+        // get speedometer label setups
+        for (KeyValues *pSpeedoItem = pGamemodeKV->GetFirstSubKey(); pSpeedoItem != nullptr;
+             pSpeedoItem = pSpeedoItem->GetNextKey())
+        {
+            const char *speedoname = pSpeedoItem->GetName();
+            SpeedometerLabel *pLabel = GetLabelFromName(speedoname);
+            KeyValues *pSpeedoKV = pGamemodeKV->FindKey(speedoname);
+            if (!pLabel || !pSpeedoKV)
+                continue;
+
+            pLabel->SetVisible(pSpeedoKV->GetBool("visible", false));
+            if (!pLabel->IsVisible())
+                continue;
+
+            int colorize = pSpeedoKV->GetInt("colorize", 0);
+            if (colorize == SPEEDOMETER_COLORIZE_RANGE)
+            {
+                KeyValues *pRangesKV = pSpeedoKV->FindKey("ranges");
+                if (!pRangesKV)
+                    continue;
+
+                RangeList *m_pRangeList = new RangeList();
+                for (KeyValues *pRangeItem = pRangesKV->GetFirstSubKey(); pRangeItem != nullptr;
+                     pRangeItem = pRangeItem->GetNextKey())
+                {
+                    Range_t range;
+                    range.min = pRangeItem->GetInt("min", 0);
+                    range.max = pRangeItem->GetInt("max", 0);
+                    range.color = pRangeItem->GetColor("color");
+                    m_pRangeList->AddToTail(range);
+                }
+                pLabel->SetRangeList(m_pRangeList);
+            }
+            pLabel->SetColorizeType(colorize);
+            pLabel->SetUnitType(pSpeedoKV->GetInt("units", 0));
```

Comment:
Likewise here, this can be under a `LoadFromKV(KeyValues *pIn)` method for the label class. This would get rid of the need to store the RangeList as a pointer (can be a class member object instead).